RestaurantReservation.Db/Repositories/CustomerRepository.cs

```diff
@@ -0,0 +1,69 @@
+﻿using Microsoft.EntityFrameworkCore;
+using RestaurantReservation.Db.DataModels;
+using RestaurantReservation.Db.Utilities;
+
+namespace RestaurantReservation.Db.Repositories
+{
+    public class CustomerRepository : IRepositoryServices<Customer,string>
+    {
+        private readonly RestaurantReservationDbContext _dbContext;
+
+        public CustomerRepository(RestaurantReservationDbContext dbContext)
+        {
+            _dbContext = dbContext;
+        }
+
+        public async Task<OperationResult<string>> AddAsync(Customer customer)
+        {
+            try
+            {
+                await _dbContext.Customers.AddAsync(customer);
+                await _dbContext.SaveChangesAsync();
+
+                return OperationResult<string>.SuccessResult($"id:{customer.Id}");
+            }
+            catch (Exception ex)
```

Comment:
any reason why you to a try catch here but not in other methods (delete, retrieve nor update)?